### PR TITLE
Restructuring directories

### PR DESCRIPTION
**Pull request by Bartekus**
Here is the version that has all the directories restructured to more closely reassemble the redux pattern. It will serve as starting point for structor.

### CODE REVIEW

**Comment by mattkrick** [comment]
IIRC `UPDATE_LOCATION` is used as kinda a hack for SSR, otherwise the router freaks out. If you make a PR that can do things the "right" way, i'm totally open to it. 

WRT functional components, currently there's no reason to use them because the current version of react doesn't optimize them. What's more, HMR can't handle them https://github.com/gaearon/babel-plugin-react-transform/issues/57

**Comment by mattkrick** [comment]
@Bartekus this encompasses #69  and #64, right? what are your thoughts about node-config? Since this is such a large PR I'm up for merging this & then we can discuss node-config later if you want.

LGTM, gimme the word & I'll merge & take lead on upgrading to the latest webpack & babel, as I know those are definitely needed

**Comment by Bartekus** [comment]
I'm all at your disposal Matt, and yes this branch adds the restructuring however if you don't want that yet the other pull has all the good stuff minus the restructuring. As to the upgrades, I've started that on another branch, I just gotta do more tests and I'll submit the pull. Depending on which branch you merge I'll fork from there and apply the upgrades.

File: webpack/webpack.config.dev.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ import webpack from 'webpack';
 import cssModulesValues from 'postcss-modules-values';
 
 const root = process.cwd();
-const clientInclude = [path.join(root, 'src', 'client'), path.join(root, 'src', 'universal')];
-const globalCSS = path.join(root, 'src', 'universal', 'styles','global');
+const clientInclude = path.join(root, 'src');
```
Comment (mattkrick) [inline webpack/webpack.config.dev.js]: This includes the server directory, that's no good.

**Comment by mattkrick** [comment]
just 1 fix, see comments

**Comment by Bartekus** [comment]
@mattkrick I've specified each folder separately and excluded the server so it should be all good!

**Pull request closed without merging**